04_fep/05_ATM/01-ATM.ipynb

```diff
@@ -0,0 +1,481 @@
+    ")\n",
+    "equilibrate_process = BSS.Process.OpenMM(minimised, equilibration)\n",
+    "equilibrate_process.start()\n",
+    "equilibrate_process.wait()\n",
+    "equilibrated = equilibrate_process.getSystem(block=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d2da0d7-f7ee-45de-9af5-a54b2f03ba4c",
+   "metadata": {},
+   "source": [
+    "The system has now been minimised and equilibrated as a vanilla system, that is without the ATMForce present. \n",
+    "\n",
+    "We now need to introduce the ATMForce to the system; this introduction needs to be gradual, otherwise our simulations will crash.\n",
+    "\n",
+    "This gradual introduction can be done by annealing the system to a lambda value of 0.5. Starting from lambda=0, the annealing protocol simulates a series of windows in which the value of lambda is gradually increased. In this case we will use 10 annealing cycles with a total runtime of 100ps, meaning that the value of lambda is increased by a value of 0.05 every 10ps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "49bc267d-4578-407c-bac6-93761f9cea9f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "annealing = BSS.Protocol.ATMAnnealing(\n",
+    "    data=atm_data,\n",
+    "    core_alignment=True,\n",
+    "    restraint=ca,\n",
+    "    com_distance_restraint=True,\n",
+    "    runtime=\"100ps\",\n",
+    "    anneal_numcycles=10,\n",
+    ")\n",
+    "annealing_process = BSS.Process.OpenMM(equilibrated, annealing, platform=\"CUDA\")\n",
+    "annealing_process.start()\n",
+    "annealing_process.wait()\n",
+    "annealed = annealing_process.getSystem(block=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a2ae2454-ff53-44df-843c-cf220ace99a3",
+   "metadata": {},
+   "source": [
+    "Finally, we need to perform a post-annealing equilibration. This equilibration step is performed at a lambda value of 0.5, and is designed to introduce the free ligand to the protein-ligand complex without actually performing a swap. This step dramatically increases the stability of production simulations in which `ligand_free' is in the binding site (i.e. those with a lambda value greater than 0.5) by decreasing the probability of atom overlaps occurring when the ligand is placed in the binding site."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e8b14066-c1a5-4e20-a492-75858031711b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "post_anneal_equilibration = BSS.Protocol.ATMEquilibration(\n",
+    "    data=atm_data,\n",
+    "    core_alignment=True,\n",
+    "    restraint=ca,\n",
+    "    com_distance_restraint=True,\n",
+    "    use_atm_force=True,\n",
+    "    lambda1=0.5,\n",
+    "    lambda2=0.5,\n",
+    "    runtime=\"100ps\",\n",
+    ")\n",
+    "post_anneal_equilibration_process = BSS.Process.OpenMM(\n",
+    "    annealed, post_anneal_equilibration, platform=\"CUDA\"\n",
```

Comment:
same comments as above

Reply:
cut to 1ps

Reply:
remind users to use 100 ps if they want to adapt the demo for actual production :-)